page/performance/optimize-selectors.md

```diff
@@ -7,19 +7,28 @@
 
 Selector optimization is less important than it used to be, as more browsers implement `document.querySelectorAll()` and the burden of selection shifts from jQuery to the browser. However, there are still some tips to keep in mind.
 
-## ID-Based Selectors
+## Save calls to `querySelectorAll`
 
-Beginning your selector with an ID is always best.
+```
+// If in your HTML there are 2 .container with 5 div in each,
+// this line will call querySelectorAll 13 times (1 + 2 + 2*5).
+$( ".container" ).children( "div" ).find( ".robotarm" );
 
+// Against only 1 call with this:
+$( ".container div .robotarm" );
 ```
-// Fast:
-$( "#container div.robotarm" );
 
-// Super-fast:
-$( "#container" ).find( "div.robotarm" );
+## Try to segregate the nonstandard parts of your selection
+
+When you select elements, jQuery will call `querySelectorAll` with your selection. If `querySelectorAll` throws an error, jQuery will refer to its Sizzle engine. So, if you are using at least one of these nonstandard pseudo-classes: `:contains()`, `:has`, `:even`, `:submit`, etc. You will not take advantage of the native `querySelectorAll`.
+
 ```
+// A long selection with nonstandard pseudo-classes inside:
```

Comment:
Maybe we can remove the colon at the end of this comment and the next for consistency. This isn't very important, though.